Assets/BossRoom/Scripts/Shared/Net/ConnectionManagement/SessionManager.cs

```diff
@@ -0,0 +1,277 @@
+        /// <param name="clientGUID">This is the clientGUID that is unique to this client and persists accross multiple logins from the same client</param>
+        /// <param name="sessionPlayerData">The player's initial data</param>
+        /// <returns></returns>
+        public ConnectStatus OnClientApprovalCheck(ulong clientId, string clientGUID, T sessionPlayerData)
+        {
+            ConnectStatus gameReturnStatus = ConnectStatus.Success;
+
+            //Test for Duplicate Login.
+            if (m_ClientData.ContainsKey(clientGUID))
+            {
+                bool isReconnecting = false;
+
+                // If another client is connected with the same clientGUID
+                if (m_ClientData[clientGUID].IsConnected)
+                {
+                    if (Debug.isDebugBuild)
+                    {
+                        Debug.Log($"Client GUID {clientGUID} already exists. Because this is a debug build, we will still accept the connection");
+
+                        // If debug build, accept connection and manually update clientGUID until we get one that either is not connected or that does not already exist
+                        while (m_ClientData.ContainsKey(clientGUID) && m_ClientData[clientGUID].IsConnected) clientGUID += "_Secondary";
+
+                        if (m_ClientData.ContainsKey(clientGUID) && !m_ClientData[clientGUID].IsConnected)
+                        {
+                            // In this specific case, if the clients with the same GUID reconnect in a different order than when they originally connected,
+                            // they will swap characters, since their GUIDs are manually modified here at runtime.
+                            isReconnecting = true;
+                        }
+                    }
+                    else
+                    {
+                        gameReturnStatus = ConnectStatus.LoggedInAgain;
+                    }
+                }
+                else
+                {
+                    isReconnecting = true;
+                }
+
+                // Reconnecting. Give data from old player to new player
+                if (isReconnecting)
+                {
+                    // Update player session data
+                    sessionPlayerData = m_ClientData[clientGUID];
+                    sessionPlayerData.ClientID = clientId;
+                    sessionPlayerData.IsConnected = true;
+                }
+            }
+
+            //Populate our dictionaries with the SessionPlayerData
+            if (gameReturnStatus == ConnectStatus.Success)
+            {
+                m_ClientIDToGuid[clientId] = clientGUID;
+                m_ClientData[clientGUID] = sessionPlayerData;
+            }
+
+            return gameReturnStatus;
+        }
+
+        public string GetPlayerGUID(ulong clientID)
```

Comment:
never used?

Reply:
indeed, not used anymore, will remove it